test/e2e/nutanix/multi-subnet.go

```diff
@@ -0,0 +1,507 @@
+				seenSubnets[identifier] = true
+			}
+		}
+	}
+}
+
+// validateNetworkConnectivity checks that machines can communicate across subnets
+func validateNetworkConnectivity(nodes *corev1.NodeList) {
+	By("validating network connectivity across subnets")
+
+	if len(nodes.Items) < 2 {
+		Skip("skipping connectivity test - need at least 2 nodes")
+		return
+	}
+
+	subnetNodes := make(map[string][]corev1.Node)
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == internalIPType {
+				subnetNodes[address.Address] = append(subnetNodes[address.Address], node)
+				break
+			}
+		}
+	}
+
+	if len(subnetNodes) < 2 {
+		Skip("skipping connectivity test - nodes not distributed across multiple subnets")
+		return
+	}
+
+	By(fmt.Sprintf("found nodes distributed across %d different subnets", len(subnetNodes)))
+}
+
+// validateMachineStatusConsistency checks that machine status reflects proper multi-subnet setup
+func validateMachineStatusConsistency(machines *machinev1beta1.MachineList) {
+	By("validating machine status consistency with multi-subnet configuration")
+
+	for _, machine := range machines.Items {
+		Expect(machine.Status.Phase).ToNot(BeNil(), "machine %s phase should be set", machine.Name)
+
+		if isRunningMachine(machine) {
+			hasInternalIP := false
+			addressCount := countInternalIPs(machine)
+
+			for _, address := range machine.Status.Addresses {
+				if address.Type == internalIPType {
+					hasInternalIP = true
+					Expect(address.Address).ToNot(BeEmpty(),
+						"address should not be empty for machine %s", machine.Name)
+				}
+			}
+
+			Expect(hasInternalIP).To(BeTrue(),
+				"running machine %s should have at least one InternalIP", machine.Name)
+
+			spec, err := ProviderSpecFromRawExtension(machine.Spec.ProviderSpec.Value)
+			Expect(err).NotTo(HaveOccurred())
+
+			if len(spec.Subnets) > 1 {
+				Expect(addressCount).To(BeNumerically(">=", len(spec.Subnets)),
+					"machine %s with %d subnets should have at least %d internal IP addresses",
+					machine.Name, len(spec.Subnets), len(spec.Subnets))
+			}
+		}
+	}
+}
+
+// validateNodeAnnotations checks node annotations for multi-subnet configuration
+func validateNodeAnnotations(nodes *corev1.NodeList, machineNetworks []configv1.NutanixFailureDomain) {
+	By("checking node annotations related to multi-subnet networking")
+
+	for _, node := range nodes.Items {
+		if cidrsJSON, exists := node.Annotations[ovnHostCIDRsAnnotation]; exists && cidrsJSON != "" {
+			var cidrs []string
+			err := json.Unmarshal([]byte(cidrsJSON), &cidrs)
+			Expect(err).NotTo(HaveOccurred(),
+				"failed to parse %s annotation for node %s", ovnHostCIDRsAnnotation, node.Name)
+
+			if hasMultiSubnetConfiguration(machineNetworks) {
+				Expect(len(cidrs)).To(BeNumerically(">=", 1),
+					"node %s should have at least one CIDR in multi-subnet setup", node.Name)
+			}
+		}
+	}
+}
+
+// runAllBasicValidations runs all basic validation functions
```

Comment:
Why add these functions? Feels like unnecessary layer of abstraction. Why are some checks Basic and some Advanced?